--- origsrc/postgresql-9.3.5/src/backend/Makefile	2014-07-21 21:10:42.000000000 +0200
+++ src/postgresql-9.3.5/src/backend/Makefile	2014-08-16 23:24:11.071321900 +0200
@@ -62,18 +62,10 @@ endif
 
 ifeq ($(PORTNAME), cygwin)
 
-postgres: $(OBJS) postgres.def libpostgres.a
-	$(DLLTOOL) --dllname $@$(X) --output-exp $@.exp --def postgres.def
-	$(CC) $(CFLAGS) $(LDFLAGS) $(LDFLAGS_EX) -o $@$(X) -Wl,--base-file,$@.base $@.exp $(call expand_subsys,$(OBJS)) $(LIBS)
-	$(DLLTOOL) --dllname $@$(X) --base-file $@.base --output-exp $@.exp --def postgres.def
-	$(CC) $(CFLAGS) $(LDFLAGS) $(LDFLAGS_EX) -Wl,--stack,$(WIN32_STACK_RLIMIT) -o $@$(X) $@.exp $(call expand_subsys,$(OBJS)) $(LIBS)
-	rm -f $@.exp $@.base
+postgres : $(OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(LDFLAGS_EX) $(export_dynamic) $(call expand_subsys,$^) $(LIBS) -o $@  -Wl,--stack,$(WIN32_STACK_RLIMIT) -Wl,--export-all-symbols -Wl,--out-implib=libpostgres.a
 
-postgres.def: $(OBJS)
-	$(DLLTOOL) --export-all --output-def $@ $(call expand_subsys,$^)
-
-libpostgres.a: postgres.def
-	$(DLLTOOL) --dllname postgres.exe --def postgres.def --output-lib $@
+libpostgres.a: postgres
 
 endif # cygwin
 
--- origsrc/postgresql-9.3.5/src/makefiles/Makefile.cygwin	2014-07-21 21:10:42.000000000 +0200
+++ src/postgresql-9.3.5/src/makefiles/Makefile.cygwin	2014-08-16 23:24:11.091322000 +0200
@@ -1,4 +1,3 @@
-DLLTOOL= dlltool
 # src/makefiles/Makefile.cygwin
 ifdef PGXS
 BE_DLLLIBS= -L$(libdir) -lpostgres
