diff -uNr postgresql_orig/src/test/regress/parallel_schedule postgresql/src/test/regress/parallel_schedule
--- postgresql_orig/src/test/regress/parallel_schedule	2014-03-18 16:25:50.031441500 +0100
+++ postgresql/src/test/regress/parallel_schedule	2014-03-18 18:07:45.402220700 +0100
@@ -78,7 +78,8 @@
 # ----------
 # Another group of parallel tests
 # ----------
-test: select_into select_distinct select_distinct_on select_implicit select_having subselect union case join aggregates transactions random portals arrays btree_index hash_index update namespace prepared_xacts delete
+test: select_into select_distinct select_distinct_on select_implicit select_having subselect union case join aggregates transactions random portals arrays btree_index hash_index update namespace delete
+# prepared_xacts
 
 # ----------
 # Another group of parallel tests
